internal/cmdutil/errors.go

```diff
@@ -43,6 +45,26 @@ func HandleError(err error) error {
 
 	perr, ok := err.(*planetscale.Error)
 	if !ok {
+		// TODO(fatih): fix the return type in our API.
+		// authErrorResponse represents an error response from the API
+		type authErrorResponse struct {
+			Error            string `json:"error"`
+			ErrorDescription string `json:"error_description"`
+			State            string `json:"state"`
+		}
+
+		errorRes := &authErrorResponse{}
+		mErr := json.Unmarshal([]byte(err.Error()), errorRes)
+		if mErr != nil {
+			// return back original error (not *mErr*). Looks like the error is
+			// not an authentication error
+			return err
+		}
+
+		if errorRes.Error == "invalid_token" {
```

Comment:
Is an invalid token also due to an expired token? Are there other cases where the token is returned as invalid?

Reply:
It actually has three states, but that's not exposed in the API:  

``` 
invalid_token:
  revoked: "The access token was revoked"
  expired: "The access token expired"
  unknown: "The access token is invalid"
``` 

This is the error body we receive:

```
{
  "error":"invalid_token",
  "error_description":"The access token is invalid",
  "state":"unauthorized"
}
```

I couldn't find the place where we return this error message though. I think Rails handles it implicitly. I'll have to sync with the team later in my evening to understand how it decides what to return.